### ISSUE TITLE
Using Python's Requests module in Pygbag for GET / POST HTTP requests

### ISSUE THREAD

**Issue by tank-king** [pygame-web/pygbag#46]
Requests is one of the most popular and easy to use HTTP library with python. Adding support for it in pygbag would be great! Also, some requests take long time to fetch data, and in such cases, use of the javascript promise (async) API or something similar to mutithreading comes handy. Having support for such a feature would be useful too!

Sample GET request:
```py
import requests
from pprint import pprint

pprint(requests.get('https://api.github.com').json())
```

Syntax for GET and POST requests:

```py
import requests

# GET request
site = 'https:// [site-name] / get_endpoint'
parameters = {'param1': '1', 'param2': '2'}
response = requests.get(site, params=parameters)
print(response.text)

# POST request
site = 'https:// [site-name] / post_endpoint'
parameters = {'param1': '1', 'param2': '2'}
response = requests.post(site, data=parameters)
print(response.text)
```

**Comment by pmp-p** [comment]
i think the GET/POST example should use fetch ( and if using raw json the solutions available here
https://stackoverflow.com/questions/29775797/fetch-post-json-data )
should be adapted to async for python with samples for classic consumers :
  - green threads ( specific to pygbag with aio )
  - js like events ( callbacks )
  - pygame user custom events ( event loop )

**Comment by STG-7** [comment]
I am creating a game using pygbag and once the game is over, I want to store the user's score in DB for leaderboards. Initially I tried using requests and though it works in a standalone pygame, the module doesn't load in pygbag even though I have put the requests wheel in the same directory as my game's main.py. Debugging the game provides following insights:
Code:

import pygame, random, asyncio, time, sys
from sys import exit, path
#import urllib.request
#import urllib.parse
sys.path.append(".")
import requests

def postScore(score):
    print("posting score")
    -- post the score to the score API

Setup:
game/main.py
game/requests-2.28.2-py3-none-any.whl

Debug error:
sys._emscripten_info(emscripten_version=(3, 1, 33), runtime='Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/110.0', pthreads=False, shared_memory=False)
pygame-ce 2.1.4.dev1 (SDL 2.24.2, Python 3.11.2)
284: assets found : 0
341: EventTarget delayed by loader
1020: import scan filename='<stdin>' len(code)=6419 ['requests']
1044: requesting module want='requests' for file='<stdin>' 
1063: repo['-CDN-']='http://localhost:8000/archives/repo/' does not provide want='requests'
1291: {req=} in {cls.ignore=} or sys.modules
646: starting shell
996: 209 lines queued for async eval
going interactive
643: TODO detect input/print to select repl debug
181: FIXME dependency table for manually built modules
1202: want='requests' found : False
PKG NOT FOUND : want='requests', resume=<function <module> at 0x5a80c0>, ex=ModuleNotFoundError("No module named 'requests'")
>>

How can I add a module wheel to my game?

**Comment by pmp-p** [comment]
> How can I add a module wheel to my game?

you must unpack wheels with a zip tool next to the main.py
But requests will NOT work on web as all I/O must be asynchronous.

find or write an async module to deal with remote I/O.

Writing an async one and keeping it minimalist is the best way since it will work on desktop too. One was started here https://github.com/pygame-web/pygbag/blob/main/pygbag/support/cross/aio/fetch.py but not finished/tested.

Personnally i don't like the fact that file,http and (web)sockets based operations are separated. I think everything should use open() and a file interface "à la plan9"

**Comment by DeadshotLegend** [comment]
The fetch is working for me. This is great. I will keep testing.
One change I had to do post JSON data. I had to add headers in the request object:
var request = new Request(url, {headers: {'Accept': 'application/json','Content-Type': 'application/json'},
        method: 'POST', 
        body: JSON.stringify(data)});
I haven't attempted to change the code in repository above. Not sure if I have access, but you might want to make this change.
